File: src/PIMobile/Entite/Location.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package PIMobile.Entite;

import com.codename1.l10n.SimpleDateFormat;
import java.util.Date;

/**
 *
 * @author hatem
 */
public class Location {
    private int id;
    private Date date_location;
    private Date date_retour;
    private Utilisateur locataire;
    private Livre livre;
    private String etat;

    public Location() {
    }

    public Location(Date date_location, Date date_retour, Utilisateur locataire, Livre livre, String etat) {
        this.date_location = date_location;
        this.date_retour = date_retour;
        this.locataire = locataire;
        this.livre = livre;
        this.etat = etat;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getDate_location() {
        return date_location;
    }

    public void setDate_location(Date date_location) {
        this.date_location = date_location;
    }

    public Date getDate_retour() {
        return date_retour;
    }

    public void setDate_retour(Date date_retour) {
        this.date_retour = date_retour;
    }

    public Utilisateur getLocataire() {
        return locataire;
    }

    public void setLocataire(Utilisateur locataire) {
        this.locataire = locataire;
    }

    public Livre getLivre() {
        return livre;
    }

    public void setLivre(Livre livre) {
        this.livre = livre;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    @Override
    public String toString() {
         SimpleDateFormat sfd = new SimpleDateFormat("dd-MM-yyyy");

        
        return "Location{" + "date_location=" + sfd.format(date_location) + ", date_retour=" + sfd.format(date_retour) + ", livre=" + livre.getTitre() + " Prix= " +livre.getPrix()+'}';
    }
    
}

File: src/BibliothequeBundle/Entity/Location.php
<?php

namespace BibliothequeBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Location
 *
 * @ORM\Table(name="location")
 * @ORM\Entity(repositoryClass="BibliothequeBundle\Repository\LocationRepository")
 */
class Location
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_location", type="date", nullable=false)
     */
    private $dateLocation;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_retour", type="date", nullable=false)
     */
    private $dateRetour;


    /**
     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\User")
     * @ORM\JoinColumn(name="locataire", referencedColumnName="id")
     */
    private $Locataire;

    /**
     * @ORM\ManyToOne(targetEntity="Livre")
     * @ORM\JoinColumn(name="livre", referencedColumnName="id")
     */
    private $Livre;

    /**
     * @var string
     *
     * @ORM\Column(name="Etat", type="string", length=255, nullable=false)
     */
    private $etat;

    /**
     * Get id
     *
     * @return integer
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set dateLocation
     *
     * @param \DateTime $dateLocation
     *
     * @return Location
     */
    public function setDateLocation($dateLocation)
    {
        $this->dateLocation = $dateLocation;

        return $this;
    }

    /**
     * Get dateLocation
     *
     * @return \DateTime
     */
    public function getDateLocation()
    {
        return $this->dateLocation;
    }

    /**
     * Set dateRetour
     *
     * @param \DateTime $dateRetour
     *
     * @return Location
     */
    public function setDateRetour($dateRetour)
    {
        $this->dateRetour = $dateRetour;

        return $this;
    }

    /**
     * Get dateRetour
     *
     * @return \DateTime
     */
    public function getDateRetour()
    {
        return $this->dateRetour;
    }

    /**
     * Set locataire
     *
     * @param \AppBundle\Entity\User $locataire
     *
     * @return Location
     */
    public function setLocataire(\AppBundle\Entity\User $locataire = null)
    {
        $this->Locataire = $locataire;

        return $this;
    }

    /**
     * Get locataire
     *
     * @return \AppBundle\Entity\User
     */
    public function getLocataire()
    {
        return $this->Locataire;
    }

    /**
     * Set livre
     *
     * @param \BibliothequeBundle\Entity\Livre $livre
     *
     * @return Location
     */
    public function setLivre(\BibliothequeBundle\Entity\Livre $livre = null)
    {
        $this->Livre = $livre;

        return $this;
    }

    /**
     * Get livre
     *
     * @return \BibliothequeBundle\Entity\Livre
     */
    public function getLivre()
    {
        return $this->Livre;
    }

    /**
     * Set etat
     *
     * @param string $etat
     *
     * @return Location
     */
    public function setEtat($etat)
    {
        $this->etat = $etat;

        return $this;
    }

    /**
     * Get etat
     *
     * @return string
     */
    public function getEtat()
    {
        return $this->etat;
    }
}
